xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/CopyJob.java

```diff
@@ -90,4 +66,16 @@ protected boolean atomicOperation(DocumentReference source, DocumentReference ta
     {
         return this.modelBridge.copy(source, target);
     }
+
+    @Override
+    protected <T extends BeginFoldEvent & CancelableEvent> T getBeginEvent()
```

Comment:
Since the type of begin event and end event depend on the class implementing then and now on the code calling the method, you should declare those two type in the class definition.